strawberry/field.py

```diff
@@ -33,6 +35,36 @@ def __init__(self, field_definition: FieldDefinition):
             metadata=None,
         )
 
+    def __call__(self, resolver: Callable) -> Callable:
+        """Migrate the field definition to the resolver"""
+
+        # TODO: This does not copy the data and now two objects share the same
+        #       FieldDefinition. Presumably this field will be GC-ed soon
+        #       though?
+        field_definition = self._field_definition
+        # Note: field_definition.name is finalized in type_resolver._get_fields
+
+        resolver_name = to_camel_case(resolver.__name__)
+
+        field_definition.origin_name = resolver_name
+
+        field_definition.origin = resolver
+        field_definition.base_resolver = resolver
+        field_definition.arguments = get_arguments_from_resolver(resolver)
+
+        field_definition.type = get_return_annotation(field_definition)
+
+        try:
```

Comment:
We probably want to use something along the lines of
```python
if inspect.ismethod(resolver) and resolver.__self__ is cls:
    ...
```

There are other things that could, in theory, cause an AttributeError?

Reply:
Also, does this work with the `hasattr(field, "_field_definition")` in `type_resolver._get_fields`?

Reply:
> There are other things that could, in theory, cause an AttributeError?

maybe, not sure when it would happen though :)

> Also, does this work with the hasattr(field, "_field_definition") in type_resolver._get_fields?

yup :) 

<img width="388" alt="image" src="https://user-images.githubusercontent.com/667029/91099738-c59c7900-e65b-11ea-98ff-8d2503f81b6a.png">

Reply:
Actually it [looks like](https://stackoverflow.com/questions/7891277/why-does-setattr-fail-on-a-bound-method) `setattr` fails on _all_ bound methods, not just classmethods... so I think all we need is 
```python
inspect.ismethod(resolver)
```

Reply:
> yup :)

huh very interesting... I didn't know there was a passthrough getattr of sorts on bound methods

Reply:
I wonder if we should check if a method is classmethod though, if we assign a bound that's not static or a classmethod then we might have a misusage, no?

Reply:
I think someone could pass a bound instance method as the resolver, right?

```python
class ResolverClass:
    _state: SomeState
    def resolve_func(self, info, arg) -> int:
        return self.do_something(self._state, info, arg)

resolver_class = ResolverClass()
```
```python
@strawberry.type
class Query:
    resolver_class.resolve_func here is a bound method
    query: int = strawberry.field(resolver=resolver_class.resolve_func)
```

Reply:
oh, good point. I wonder what happens in that case with self though. This is something for later I guess (we can add an issue and a test for this), not sure it is going to be common anyway 💯

Reply:
I added a test for bounded instance method resolvers... Is there anything further that we wanted to investigate in another issue?